### PR TITLE
Consolidate webasset bundle to single chunks

### PR DESCRIPTION
**Pull request by avatus**
Refer to the ["Why" section of this RFD](https://github.com/gravitational/teleport/pull/35757/files#diff-dd7eef5f499a7ee372ff250562b5347616bd13b343cba2654506b8f3c652a907R17) for info on the issue solved with this PR.

This PR will configure vite to pipe all code chunks together. So techincally, this isn't removing code splitting, but in reality just not splitting the code. The benefits of this way are
1. we do not have to remove any of our dynamically imported code as it will just import it all at build time
2. in the future, we can take advantage of the code splitting function (right now its essentially an empty anonymous function) to have more fine grained control over the code splitting in the future. 

This ignores fonts/images/wasm, and only affects javascript and css files. We will now have a single javascript and css file load on initial page load. This will increase latency slightly as there is a larger file to download but it will remove any other network requests for javascript files after (since it has the entire bundle) and *usually* the javascript file and css file will be cached by the browser anyway so this initial latency increase should only happen once per version (or however their browser may be set up).

Quick loom videos showing the error and fix
https://www.loom.com/share/bcae5d79b14a4e6e8df8e0603ace0f04
https://www.loom.com/share/2ce5386941c849dca6a142089d927aa3

### CODE REVIEW

**Comment by avatus** [comment]
Updated with some loom videos showing it work/not-work 
https://www.loom.com/share/bcae5d79b14a4e6e8df8e0603ace0f04
https://www.loom.com/share/2ce5386941c849dca6a142089d927aa3

Please help me break this!

**Comment by avatus** [comment]
I did toy with the idea of keeping the hashing for `index.js` and `index.css` ONLY, just to make sure those are "in sync" since they are loaded at the same time. I think I went without just to make sure something is always loaded rather than a rare case that the instant one loads it load balances to the other version. Slightly broken css might not be as bad as "page didn't load at all, refresh"

**Comment by ryanclark** [comment]
> I did toy with the idea of keeping the hashing for `index.js` and `index.css` ONLY, just to make sure those are "in sync" since they are loaded at the same time. I think I went without just to make sure something is always loaded rather than a rare case that the instant one loads it load balances to the other version. Slightly broken css might not be as bad as "page didn't load at all, refresh"

`index.css` only contains the fonts so it'll very rarely change

File: web/packages/build/vite/config.ts
Comment (ryanclark) [inline web/packages/build/vite/config.ts:75]: Can we remove `React.lazy` and all the default exports instead? Otherwise we're wrapping the loading of all route components in a promise for no reason

Comment (avatus) [reply web/packages/build/vite/config.ts:75]: Removed react lazy on everything except access graph/assist. Was getting really weird "React Markdown" errors unless it was lazy loaded. like 40+ of them during testing. I'll keep inspecting but for now, every other lazy is removed

Comment (avatus) [reply web/packages/build/vite/config.ts:75]: We could something like with `manualChunks` to programmatically choose which chunks are split and remove hashing that way for those specifically, or we can keep this line to handle the assist/access graph and call it a day. I dont know much about assist's internals I think to know what to chunk out

Comment (avatus) [reply web/packages/build/vite/config.ts:75]: e update if we stick with these changes https://github.com/gravitational/teleport.e/pull/3566

**Comment by avatus** [comment]
Another loom of a "upgrade in process" with multiple sessions. There was an error that showed up but its because my second proxy couldn't reach the auth server (which is running with my first proxy) during an upgrade but thats not related to my webassets https://www.loom.com/share/1cd8dd618b054b63ac2462387ba32b07

**Comment by avatus** [comment]
Another way to help review this is to build the UI and just poke around to make sure everything works still. I made large changes to imports of features and while I tried to check most it'd be nice to have another set of eyes to check for regressions

File: web/packages/build/vite/config.ts
Comment (ibeckermayer) [inline web/packages/build/vite/config.ts:74]:
```suggestion
            // This prevents code splitting.
```

File: web/packages/build/vite/config.ts
Comment (ibeckermayer) [inline web/packages/build/vite/config.ts:80]: > We will keep the javascript file hashed

doesn't this contradict the previous comment of

> removes hashing from our entry point file

?

File: web/packages/teleport/src/Integrations/Enroll/AwsOidc/index.ts
Comment (ibeckermayer) [inline web/packages/teleport/src/Integrations/Enroll/AwsOidc/index.ts:19]: Why are we keeping this one?

File: web/packages/teleport/src/LocksV2/NewLock/LockCheckout/index.ts
Comment (ibeckermayer) [inline web/packages/teleport/src/LocksV2/NewLock/LockCheckout/index.ts:19]: And this one?

**Review by ibeckermayer — COMMENTED** [review]
The linked-to RFD is still open, and this change appears to be talked about as an aside at the end. Was it decided somewhere that this is the approach that we're going with? If yes, have we characterized the performance impact?

File: web/packages/build/vite/config.ts
Comment (avatus) [reply web/packages/build/vite/config.ts]: sorry, an older version and I left the comments in. thanks

File: web/packages/teleport/src/Teleport.tsx
Comment (ryanclark) [inline web/packages/teleport/src/Teleport.tsx:43]: These are all default exports

File: web/packages/teleport/src/TrustedClusters/index.ts
Comment (ryanclark) [inline web/packages/teleport/src/TrustedClusters/index.ts:20]: Can we use this PR to clean these up all the way through to the component?

That way we can have a consistent

```typescript
export { TrustedClusters } from './TrustedClusters';
```

I know it's slightly increasing the scope of this PR, but we'd never get round to the cleanup otherwise

Feel free to ignore

File: web/packages/teleport/src/features.tsx
Comment (ryanclark) [inline web/packages/teleport/src/features.tsx:55]: Still default

**Review by ryanclark — APPROVED** [review]
Approving as it LGTM, pending the last default export changes

**Review by rudream — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

**Pull request merged**
